build.gradle

```diff
@@ -180,10 +180,7 @@ configure(opentelemetryProjects) {
 
     dependencies {
 
-        implementation platform(boms.grpc),
-                platform(boms.guava),
-                platform(boms.protobuf),
-                platform(boms.zipkin_reporter)
+        implementation platform(boms.grpc)
```

Comment:
I don't think we want enforce versions for all grpc components for all artifacts. But probably this is a (serious) issue that will only be resolved with #575. That is, even though we "only" depend on grpc-context, we force every user of the API to only use grpc versions (including other artifacts) that can work together with our grpc-context version. BTW, if grpc also depends on the proposed "propagation.io" context, we will still have the same problem.

Reply:
Not sure I understand the latest concern. Can you elaborate why it will be the same problem?

Reply:
Any user of propagation-io-context still needs to use a propagation-io-context version compatible with the grpc version they use. It's not the same problem with BOMs but that fundamental problem remains.

Reply:
If the user uses a grpc BOM, that BOM should include propagation-io-context and would override the version declared by opentelemetry-api.

On second thought, as for this PR, I think we should just not use the GRPC bom. BOMs for ensuring that all the versions of related dependencies match. But we want to depend only on a single dependency, namely the context. So there is nothing we could be incompatible with. If the user uses a grpc bom, it will override the API grpc version, which can be bad (see first paragraph of this comment, and my previous comment), but arguably better than opentelemetry-api overriding the grpc version of a user who does not use BOMs (not sure if that's the case).

Reply:
```suggestion
```

Reply:
we should use the bom for otlp exporter probably but not for API, I think that is what you say. Also user can overwrite all the versions by using their own bom. A bom ensures that all the versions for a set of artifacts are the same, user can still overwrite the bom version.

Reply:
We use several grpc artifacts, e.g. `grpc-protobuf` and `grpc-stub`. So using bom to align their version seems like a totally sensible thing to do. The only concern that I raised and try to solve with this PR, is the unneeded dependency in `api`, which breaks api clients, such as otel auto-instrumentation agent. All other concerns regarding artifacts dependencies can be addressed separately, if they signal the actual problems.

Reply:
I agree we should use the grpc BOM for anything that actually uses grpc. But here it is applied to all artifacts, including API and SDK.

Reply:
api uses `context_prop` which uses grpc.

Reply:
From  #575 I got the impression that we should threat grpc-context as "not actually grpc but just in the same group". Anyway, it uses a single artifact from that BOM so I think the following choices are equally valid technically:

1. Use the BOM for api. Then any consumers of api using gprc will be forced (by default; I know now this can be overriden) to use a grpc compatible with the grpc-context that the API uses.
2. Don't use the BOM for api. Then any consumers of api using grpc will use whatever version of grpc they declared. If that version is incompatible with our grpc-context, I'm not sure what will happen.

Reply:
> Don't use the BOM for api.

+1 on this. This would simplify things overall (which is exactly what we need for the API case).

Reply:
That would mean removing all shared dependency management related stuff from `api`. I cannot even write `api libraries.grpc_context` then, but have to spell it out `api "io.grpc:grpc-context:${grpcVersion}"`. Are you sure? :)